cmd/antrea-agent/agent.go

```diff
@@ -320,30 +323,46 @@ func run(o *Options) error {
 		)
 	}
 
-	var mcRouteController *mcroute.MCRouteController
+	var mcDefaultRouteController *mcroute.MCDefaultRouteController
 	var mcStrechedNetworkPolicyController *mcroute.StretchedNetworkPolicyController
+	var mcPodRouteController *mcroute.MCPodRouteController
 	var mcInformerFactory mcinformers.SharedInformerFactory
+	var clusterServiceCIDR *net.IPNet
+	var mcInformerFactoryWithOption mcinformers.SharedInformerFactory
+
 	if enableMulticlusterGW {
-		mcNamespace := env.GetPodNamespace()
-		if o.config.Multicluster.Namespace != "" {
-			mcNamespace = o.config.Multicluster.Namespace
-		}
-		mcInformerFactory = mcinformers.NewSharedInformerFactory(mcClient, informerDefaultResync)
-		gwInformer := mcInformerFactory.Multicluster().V1alpha1().Gateways()
-		ciImportInformer := mcInformerFactory.Multicluster().V1alpha1().ClusterInfoImports()
-		mcRouteController = mcroute.NewMCRouteController(
+		mcInformerFactoryWithOption = mcinformers.NewSharedInformerFactoryWithOptions(mcClient,
+			informerDefaultResync,
+			mcinformers.WithTweakListOptions(func(lo *metav1.ListOptions) {
+				lo.FieldSelector = fields.Set{"metadata.namespace": o.config.Multicluster.Namespace}.String()
+			}),
+		)
+		mcDefaultRouteController = mcroute.NewMCDefaultRouteController(
 			mcClient,
-			gwInformer,
-			ciImportInformer,
+			mcInformerFactoryWithOption,
 			ofClient,
 			ovsBridgeClient,
 			ifaceStore,
 			nodeConfig,
-			mcNamespace,
+			o.config.Multicluster.Namespace,
 			o.config.Multicluster.EnableStretchedNetworkPolicy,
 		)
+		_, clusterServiceCIDR, _ = net.ParseCIDR(o.config.Multicluster.ServiceCIDR)
```

Comment:
I am confused on this. `clusterServiceCIDR` is passed to poxier but why initialize it here?

BTW, you still plan to have a separate PR for the Endpoints selection change right? If so, could you merge the 2nd and 3rd commits to one? Otherwise, we cannot exclude the changes from the Endpoints selection commit, when reviewing this PR.

Reply:
It's used only when `enableMulticlusterGW` is true. This part will be refined after #4570 is merged.
Top 2 commits are merged.